File: src/arduinobot_py_examples/arduinobot_py_examples/simple_publisher.py
#!/usr/bin/env python3



import rclpy 
from rclpy.node import Node 
from std_msgs.msg import String

class SimplePublisher(Node):
    def __init__(self):
        super().__init__("simple_publisher_py")
        self.pub_ = self.create_publisher(String, "chatter", 10)
        self.counter_=0
        self.frequency_=3.14
        self.pub_timer_ = self.create_timer(self.frequency_,self.pubTimerCallback)
        self.get_logger().info("Object of SimplePublisher class has been created.")
        self.get_logger().info("Publishing at %f Hz"%self.frequency_)

    
    def pubTimerCallback(self):
        msg = String()
        msg.data = "[Hi there -> %d]"%self.counter_
        self.pub_.publish(msg)
        self.counter_+=1

File: src/arduinobot_cpp_examples/src/simple_publisher.cpp
/**
 * @file simple_publisher.cpp
 * @author vaibhav
 * @brief
 * @version 0.1
 * @date 2023-11-02
 *
 *
 */

#include "arduinobot_cpp_examples/simple_publisher.h"

SimplePublisher::SimplePublisher() : Node("simple_publisher_cpp"), counter_(0)
{
    pub_ = this->create_publisher<std_msgs::msg::String>("chatter", 10);

    pub_timer_ = this->create_wall_timer(std::chrono::milliseconds(314),
                                         std::bind(&SimplePublisher::pubTimerCallback, this));

    RCLCPP_INFO(this->get_logger(), "object of SimplePublisher class has beencreated.");
    RCLCPP_INFO(this->get_logger(), "published frequency: 3.14");
}

void SimplePublisher::pubTimerCallback()
{
    auto msg = std_msgs::msg::String();
    msg.data = "[HI -> " + std::to_string(counter_) + "]";
    pub_->publish(msg);
    counter_++;
}